ledger/participant-integration-api/src/main/scala/platform/apiserver/services/transaction/ApiTransactionService.scala

```diff
@@ -72,9 +72,16 @@ private[apiserver] final class ApiTransactionService private (
       logger.debug(s"Received request for transaction subscription $subscriptionId: $request")
       transactionsService
         .transactions(request.startExclusive, request.endInclusive, request.filter, request.verbose)
+        .via(logger.debugStream(transactionsLoggable))
         .via(logger.logErrorsOnStream)
     }
 
+  private def transactionsLoggable(transactions: GetTransactionsResponse): String =
+    s"Responding with transactions: ${transactions.transactions.map(_.toProtoString).mkString("[", ",", "]")}"
+
+  private def transactionTreesLoggable(trees: GetTransactionTreesResponse): String =
+    s"Responding with transaction trees: ${trees.transactions.map(_.toProtoString).mkString("[", ",", "]")}"
+
```

Comment:
I would appreciate a comment on whether we want complete transactions and transaction trees logged or should we log only a selection of fields? @mziolekda

Reply:
Thanks, it is very thoughtful of you.
Those logs end up being way to verbose when passed through proto. We only need transactionId, commandId, workflowId and offset under those very names.

Reply:
Fixed 👍